### PR TITLE
WIP - PoC phase: Create the pod for upgrade and downgrade

### PR DESCRIPTION
**Pull request by houshengbo**
This PR offers us an option to upgrade the eventing operator via another separate reconcile function.

### CODE REVIEW

File: cmd/version-controller/main.go [hunk 1]
```diff
@@ -1,13 +1,26 @@
-Copyright (c) 2015 The New York Times Company
- 
+/*
+Copyright 2018 The Knative Authors
```
Comment (mattmoor) [inline cmd/version-controller/main.go]: Incorrect copyright year:
```suggestion
Copyright 2019 The Knative Authors
```

File: config/version-controller.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+              fieldPath: metadata.name
+        - name: SYSTEM_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        - name: METRICS_DOMAIN
+          value: knative.dev/eventing-operator
+        ports:
+          - name: metrics
+            containerPort: 9090
```
Comment (mattmoor) [inline config/version-controller.yaml]: Add trailing newline:
```suggestion
            containerPort: 9090

```

Comment (mattmoor) [inline config/version-controller.yaml]: Add trailing newline:
```suggestion
            containerPort: 9090

```

File: pkg/reconciler/versioncontroller/oldschema/knativeeventing.go [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+	plural, _ := meta.UnsafeGuessKindToResource(schema.GroupVersionKind{Group: GroupName,
+		Version: SchemaVersion, Kind: Kind})
+	return plural
+}
+
+func GetOldCRD(version string) schema.GroupVersionResource {
+	plural, _ := meta.UnsafeGuessKindToResource(schema.GroupVersionKind{Group: GroupName,
+		Version: SchemaVersion, Kind: Kind})
+	return plural
+}
```
Comment (mattmoor) [inline pkg/reconciler/versioncontroller/oldschema/knativeeventing.go]: Add trailing newline:
```suggestion
}

```

File: pkg/reconciler/versioncontroller/oldschema/knativeeventing.go [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+func InstallOldCRD(version string) error {
+	// Install the old CRD based on the old version.
+	//plural, _ := meta.UnsafeGuessKindToResource(schema.GroupVersionKind{Group: GroupName,
+	//	Version: SchemaVersion, Kind: Kind})
+	return nil
+}
```
Comment (mattmoor) [inline pkg/reconciler/versioncontroller/oldschema/knativeeventing.go]: Add trailing newline:
```suggestion
}

```

Comment (mattmoor) [inline pkg/reconciler/versioncontroller/oldschema/knativeeventing.go]: Add trailing newline:
```suggestion
}

```

File: pkg/reconciler/versioncontroller/oldschema/knativeeventing.go [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+			},
+		},
+	}
+
+	_, err := clientset.ApiextensionsV1beta1().CustomResourceDefinitions().Create(crd)
+	if err != nil && apierrors.IsAlreadyExists(err) {
+		return nil
+	}
+	return err
+}
```
Comment (mattmoor) [inline pkg/reconciler/versioncontroller/oldschema/knativeeventing.go]: Add trailing newline:
```suggestion
}

```

File: pkg/reconciler/versioncontroller/oldschema/knativeeventing.go [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
```
Comment (mattmoor) [inline pkg/reconciler/versioncontroller/oldschema/knativeeventing.go]: Add trailing newline:
```suggestion
}

```

**Comment by matzew** [comment]
can u rebase, @houshengbo ?

**Pull request closed without merging**